src/app/store/zone/selectors.ts

```diff
@@ -1,13 +1,110 @@
-import { generateBaseSelectors } from "app/store/utils";
-import { ZoneMeta } from "app/store/zone/types";
-import type { Zone, ZoneState } from "app/store/zone/types";
+import { createSelector } from "@reduxjs/toolkit";
 
-const searchFunction = (zone: Zone, term: string) => zone.name.includes(term);
+import { ZONE_ACTIONS } from "./constants";
+import type {
+  ZoneActionNames,
+  ZoneGenericActions,
+  ZoneModelActions,
+  ZonePK,
+  ZoneState,
+} from "./types";
+import { ZoneMeta } from "./types";
 
-const selectors = generateBaseSelectors<ZoneState, Zone, ZoneMeta.PK>(
```

Comment:
When we land this PR we should create a followup issue to build the new generate util.